### PR TITLE
feat: static_proxy.

### PR DESCRIPTION
**Pull request by skykaka**
Generate static proxy for mapper class for the user can use the mapper without the mapper object.

Which is discussed at https://github.com/smotastic/smartstruct/pull/57#discussion_r901207690.

### CODE REVIEW

**Comment by smotastic** [comment]
Hi @skykaka 
Thank you for the contribution, 
As discussed I like the idea, so I'll make sure to check it out on the weekend

**Comment by skykaka** [comment]
Hi @smotastic 
The code is not reviewed.  Have you been busy recently, or is there some problem?

File: smartstruct/lib/src/annotations.dart [hunk 1]
```diff
@@ -4,8 +4,9 @@
 class Mapper {
   final bool useInjection;
   final bool caseSensitiveFields;
+  final bool static;
```
Comment (smotastic) [inline smartstruct/lib/src/annotations.dart:7]: Hey, only thing I'd like to discuss is the name of the configuration. Everything else looks good.
Maybe something like ``generateStaticProxy``?

**Review by smotastic — CHANGES_REQUESTED** [review]
Hey, sorry I couldn't check the changes out last week. As you assumed I was very busy. But I like the change overall, just one small suggestion in the code.

**Comment by skykaka** [comment]
Very import suggestion. The name "static" is so arbitrary. I haven't thought enough about this. The name  'generateStaticProxy' is good enough. It's very clear. I think we'd better use this name. I will finish the modification of the code in the next few days.

**Review by smotastic — APPROVED** [review]

**Pull request merged**